### ISSUE TITLE
provide API call to retrieve agent IP

### ISSUE THREAD

**Issue by cjdcordeiro** [mF2C/ResourceManagement#33]
1) discovery establishes a wireless connection with a leader (@zeineb-rejiba )

2) vpnclient starts in parallel (by itself)

3) resource categorizer will ask the discovery (/api/my-ip) for the agent's IP (@zeineb-rejiba and @ssgUPC )

4) if the discovery cannot provide an IP (empty or broken), then the res cat falls back into the VPN IP (@ssgUPC )

**Comment by zeineb-rejiba** [comment]
Some comments on this. Right now 1) is done without the leader assigning IPs to agents (i.e. dnsmasq that is usually used to do so is not part of the leader). Then, when we also add the IP assignment feature to the leader, there are two alternatives:
1)	The agent is assigned an IP by the leader and is allowed to access the internet through the leader:
Pros: The agent does not need to use a different network interface for other communications over the internet.
Cons: This is likely to cause security issues (this is why we discarded this option at the early phases of the project)
2)	The agent is assigned an IP by the leader and is NOT allowed to access the internet through the leader:
Pros: Less security risks.
Cons: The agent has to use a network interface other than the one used by the discovery for internet access (e.g. to communicate with the CAU)

**Comment by zeineb-rejiba** [comment]
The latest version of the discovery provides a GET REST call: http://discovery:46040/api/v1/resource-management/discovery/my_ip/ that returns an empty IP for now: `{'IP_address': ''}` so that step 4. above can be tested.

**Comment by zeineb-rejiba** [comment]
The rest of this feature is implemented now, it is committed in the branch `adddhcp` in Gitlab. 
The leader is using the `192.168.7.0/24` network (so it is not conflicting with the VPN's `192.168.255.0/24` network)
I've tested this whith a leader and a microagent (RPI) and it's working (the IP address is assigned and retrieved correctly). 
Pending tests with a leader and a regular agent, which will be done when I'm given access to another test machine after the holidays.

**Comment by eva6370** [comment]
Step 4 corresponds to #30

**Comment by ssgUPC** [comment]
Fixed!!!
